classes/services/queryBuilders/UserListQueryBuilder.inc.php

```diff
@@ -276,15 +309,30 @@ public function get() {
 		$q = Capsule::table('users as u')
 					->leftJoin('user_user_groups as uug', 'uug.user_id', '=', 'u.user_id')
 					->leftJoin('user_groups as ug', 'ug.user_group_id', '=', 'uug.user_group_id')
-					->where('ug.context_id','=', $this->contextId)
 					->orderBy($this->orderColumn, $this->orderDirection)
 					->groupBy('u.user_id');
 
+		// context
+		if (!empty($this->contextIds)) {
+			$q->whereIn('ug.context_id', $this->contextIds);
+		} elseif (!empty($this->contextId)) {
+			$q->where('ug.context_id', '=', $this->contextId);
+		}
+
 		// roles
 		if (!is_null($this->roleIds)) {
 			$q->whereIn('ug.role_id', $this->roleIds);
 		}
 
+		// user groups
+		if (!is_null($this->userGroupIds)) {
+			if (in_array(0, $this->userGroupIds)) {
```

Comment:
is this for the boolean case above?

Reply:
A remnant, but no longer boolean. Pass `0` to return users with _no_ groups.